### PR TITLE
zbuf.NewCompareFn use order.Which

### LINKED ISSUE #2751: change zbuf.NewCompareFn() to use order.Which

### PR DESCRIPTION
**Pull request by mattnibs**
Also change dag.Merge proc to use order.Which

Closes #2751

### CODE REVIEW

File: compiler/optimizer/parallelize.go [hunk 1]
```diff
@@ -203,16 +203,12 @@ func insertMerge(seq *dag.Sequential, layout order.Layout) error {
 	if layout.IsNil() {
 		return nil
 	}
-	var reverse bool
-	if layout.Order == order.Desc {
-		reverse = true
-	}
 	// XXX Fix this to handle multi-key merge; change dag.Merge.Reverse.
 	// See Issue #2657.
```
Comment (nwt) [inline compiler/optimizer/parallelize.go:207]:
```suggestion
```

File: compiler/ast/dag/operator.go [hunk 2]
```diff
@@ -57,9 +57,9 @@ type (
 		Args     []Assignment `json:"args"`
 	}
 	Merge struct {
-		Kind    string     `json:"kind" unpack:""`
-		Key     field.Path `json:"key"`
-		Reverse bool       `json:"reverse"`
+		Kind  string      `json:"kind" unpack:""`
+		Key   field.Path  `json:"key"`
+		Order order.Which `json:"reverse"`
```
Comment (nwt) [inline compiler/ast/dag/operator.go:62]:
```suggestion
		Order order.Which `json:"order"`
```

File: zbuf/merger.go [hunk 3]
```diff
@@ -39,11 +38,15 @@ type batch struct {
...
 }
 
-func NewCompareFn(mergeField field.Path, reversed bool) expr.CompareFn {
-	nullsMax := !reversed
-	fn := expr.NewCompareFn(nullsMax, expr.NewDotExpr(mergeField))
+func NewCompareFn(layout order.Layout) expr.CompareFn {
+	nullsMax := layout.Order == order.Asc
+	exprs := make([]expr.Evaluator, len(layout.Keys))
+	for i := range layout.Keys {
+		exprs[i] = expr.NewDotExpr(layout.Keys[i])
```
Comment (nwt) [inline zbuf/merger.go:45]:
```suggestion
	for i, key := range layout.Keys {
		exprs[i] = expr.NewDotExpr(keys)
```

File: compiler/optimizer/parallelize.go (hunk 1 above)
Comment (mattnibs) [reply compiler/optimizer/parallelize.go:207]: Multi-key merge is still not possible with this pr so the comment should remain.

File: compiler/optimizer/parallelize.go [hunk 4]
```diff
@@ -203,16 +203,12 @@ func insertMerge(seq *dag.Sequential, layout order.Layout) error {
 	if layout.IsNil() {
 		return nil
 	}
-	var reverse bool
-	if layout.Order == order.Desc {
-		reverse = true
-	}
 	// XXX Fix this to handle multi-key merge; change dag.Merge.Reverse.
```
Comment (nwt) [inline compiler/optimizer/parallelize.go:206]: `dag.Merge.Reverse` goes away in this change.
```suggestion
	// XXX Fix this to handle multi-key merge.
```

**Review by nwt — APPROVED** [review]

**Pull request merged**